posthog/caching/warming.py

```diff
@@ -80,25 +79,24 @@ def priority_insights(team: Team) -> Generator[tuple[int, Optional[int]], None,
     yield from dashboard_tiles
 
 
-@shared_task(ignore_result=True, expires=60 * 60)
+@shared_task(ignore_result=True, expires=60 * 15)
 def schedule_warming_for_teams_task():
-    team_ids = largest_teams(limit=3)
+    skew_seconds = 3
+    team_ids = largest_teams(limit=10)
 
     teams = Team.objects.filter(Q(pk__in=team_ids) | Q(extra_settings__insights_cache_warming=True))
 
     logger.info("Warming insight cache: teams", team_ids=[team.pk for team in teams])
 
-    # TODO: Needs additional thoughts about concurrency and rate limiting if we launch chains for a lot of teams at once
-
     for team in teams:
         insight_tuples = priority_insights(team)
 
-        task_groups = chain(*(warm_insight_cache_task.si(*insight_tuple) for insight_tuple in insight_tuples))
-        task_groups.apply_async()
+        # We skew the task execution to reduce queries *for a single team* running at the same time
+        group(warm_insight_cache_task.si(*insight_tuple) for insight_tuple in insight_tuples).skew(step=skew_seconds)()
```

Comment:
`chain()` seems like a more surefire way to reduce simultaneous queries for a single team, so I wonder what's the rationale for `group()` with skewing being better?